### PR TITLE
Table column - allow for values to be set from code

### PR DESCRIPTION
**Pull request by rabol**
This change allow the developer to specify the data,  as an array, that should be displayed in the table.

if no 'value' is provided, fetch from DB as normal.

### CODE REVIEW

File: src/resources/views/crud/columns/table.blade.php [hunk 1]
```diff
@@ -1,5 +1,12 @@
 @php
-	$value = data_get($entry, $column['name']);
+	if(!isset($column['value']))
+    {
+        $value = data_get($entry, $column['name']);
+    }
+    else
+    {
+        $value = $column['value'];
+    }
```
Comment (tabacitu) [inline src/resources/views/crud/columns/table.blade.php:9]: I see no reason to not support this, thanks @rabol ! I think we can write this a lot shorter though, would using the null coalescing operator work just as well?
```suggestion
        $value = $column['value'] ?? data_get($entry, $column['name']);
```

Comment (tabacitu) [reply src/resources/views/crud/columns/table.blade.php:9]: Oh I just thought of something for the maintenance team to check. 

For whoever picks up this issue: 

1) Do other columns support the `value` attribute? 
- If they do, I think we should add this.
- If they do NOT, I think we should talk about it some more before merging.

2) Which columns support the `value` attribute? If one column does, then... ALL columns should. Right?

**Review by tabacitu — CHANGES_REQUESTED** [review]

**Comment by rabol** [comment]
As backpack supports Laravel 6, that means that php 7.2 is valid and I do not remember if the ?? is supported in php 7.2

**Comment by pxpm** [comment]
@rabol AFAIK null coalescing operator (`??`) is supported since PHP 7 - https://www.php.net/manual/en/migration70.new-features.php 
No worries about it. 

AFAIK there is no `value` attribute in other columns like @tabacitu mentioned. 

I think we are reinventing the wheel here adding that functionality in columns. What should be done is adding an acessor to the model. It's for scenarios like this that Laravel introduced them I think :) https://laravel.com/docs/8.x/eloquent-mutators#defining-an-accessor

```php

public function getSomeTableAttribute() { 
    return array('your_table_array');
}

```

Let me know,
Pedro

**Comment by rabol** [comment]
I'm happy with an accessor.

**Comment by pxpm** [comment]
Thanks @rabol if so, can we close this ? 

I really appreciate your effort creating the PR. For me, you are already part of the team🥇 

Wish you the best,
Pedro

**Comment by rabol** [comment]
Yes, we can close it, but how do we get the accessor to work, new PR ?

**Comment by pxpm** [comment]
@rabol accessor is something you add in your application. Is not Backpack specific. 

If you have a field:
```php
$this->crud->addField(['name' => table', 'type' => 'table']);
```

you create an acessor in the model with:

```php
public function getTableAttribute() { 
    return array('your_table_array');
   //here $this refers to the current entry if you need access to other attributes
}
```

When loading the field, backpack will call `$entry->table` and return whatever value you return in your acessor (You need to build the array to work with table field in this scenario)

If you have any doubts I am around in gitter. 

Thanks again,
Pedro

**Comment by tabacitu** [comment]
Ugh yeah ok using a Laravel feature is even better than building it out ourselves 😂 Good catch @pxpm . @rabol check out [the docs for Laravel accessors here](https://laravel.com/docs/8.x/eloquent-mutators#accessors-and-mutators)

Cheers!

**Pull request closed without merging**